go-controller/pkg/ovn/policy_test.go

```diff
@@ -1977,6 +1978,9 @@ var _ = ginkgo.Describe("OVN NetworkPolicy Operations", func() {
 						updateNamespaceACLLogSeverity(&originalNamespace, desiredLogSeverity, desiredLogSeverity)).To(gomega.Succeed(),
 						"should have managed to update the ACL logging severity within the namespace")
 
+					// TODO: come up with something better. This smells bad.
+					time.Sleep(time.Second)
```

Comment:
@dcbw / @trozet this looks really bad. 

I've ran this locally in a loop 100 times and it sometimes failed because the new policy was created with the old logging level; clearly something is wrong. It "worked" consistently after I've added this wait.

I'll dig into the logs of a failed test execution to try to figure out what went wrong.

Reply:
On **failed** test executions I see that the policy gets created **before** the annotation update:

```
...
...
STEP: updating the namespace's ACL logging severity
I1008 13:08:40.962283  104090 ovn.go:681] Bootstrapping existing policies and cleaning stale policies took 12.737µs
STEP: Provisioning a new network policy
I1008 13:08:40.962438  104090 policy.go:794] Adding network policy networkpolicy1 in namespace namespace1
I1008 13:08:40.962453  104090 namespace.go:231] [namespace1] updating namespace
dropping table update
dropping table update
I1008 13:08:40.964787  104090 policy.go:821] ACL logging for network policy networkpolicy1 in namespace namespace1 set to deny=alert, allow=notice
...
```

While usually (**successful** executions), what happens is the opposite; the namespace update is handled **before** provisioning the new policy:

```
...
...
STEP: updating the namespace's ACL logging severity
STEP: Provisioning a new network policy
I1008 13:08:52.450733  104281 namespace.go:231] [namespace1] updating namespace
I1008 13:08:52.450885  104281 policy.go:794] Adding network policy networkpolicy1 in namespace namespace1
...
```

Is there a way we can ensure the namespace controllers have higher priority than (run before) the policy controllers ?

Reply:
@maiqueb there's no way to enforce ordering, nor should we. Events can happen at any time, in any order, and code has to deal with that. We're moving code to that model (eg, Services and hybrid-overlay are already handled this way) but it's not all there yet.

What typically happens is to exit early if the thing you need is not there yet, and when it finally happens, retry the operation. If the policy code needs the namespace but it's not there yey, ignore the event. But then when the namespace event *does* happen you figure out all the policy events that didn't succeed, but now can succeed, and retry them.